contrib/signet/getcoins.py

```diff
@@ -5,32 +5,64 @@
 
 import argparse
 import subprocess
-import requests
 import sys
+import requests
+
+DEFAULT_GLOBAL_FAUCET = 'https://signetfaucet.com/claim'
+GLOBAL_FIRST_BLOCK_HASH = '00000086d6b2636cb2a392d45edc4ec544a10024d30141c9adf4bfd9de533b53'
 
 parser = argparse.ArgumentParser(description='Script to get coins from a faucet.', epilog='You may need to start with double-dash (--) when providing bitcoin-cli arguments.')
 parser.add_argument('-c', '--cmd', dest='cmd', default='bitcoin-cli', help='bitcoin-cli command to use')
-parser.add_argument('-f', '--faucet', dest='faucet', default='https://signetfaucet.com/claim', help='URL of the faucet')
+parser.add_argument('-f', '--faucet', dest='faucet', default=DEFAULT_GLOBAL_FAUCET, help='URL of the faucet')
 parser.add_argument('-a', '--addr', dest='addr', default='', help='Bitcoin address to which the faucet should send')
 parser.add_argument('-p', '--password', dest='password', default='', help='Faucet password, if any')
 parser.add_argument('bitcoin_cli_args', nargs='*', help='Arguments to pass on to bitcoin-cli (default: -signet)')
 
 args = parser.parse_args()
 
-if args.addr == '':
-    if args.bitcoin_cli_args == []:
-        args.bitcoin_cli_args = ['-signet']
-    # get address for receiving coins
+if args.bitcoin_cli_args == []:
+    args.bitcoin_cli_args = ['-signet']
+
+
+def bitcoin_cli(rpc_command_and_params):
+    argv = [args.cmd] + args.bitcoin_cli_args + rpc_command_and_params
     try:
-        args.addr = subprocess.check_output([args.cmd] + args.bitcoin_cli_args + ['getnewaddress', 'faucet', 'bech32']).strip()
+        return subprocess.check_output(argv).strip().decode()
     except FileNotFoundError:
         print('The binary', args.cmd, 'could not be found.')
-        exit()
+        exit(1)
+    except subprocess.CalledProcessError:
+        cmdline = ' '.join(argv)
+        print(f'-----\nError while calling "{cmdline}" (see output above).')
+        exit(1)
+
+
+if args.faucet.lower() == DEFAULT_GLOBAL_FAUCET:
+    # Get the hash of the block at height 1 of the currently active signet chain
+    curr_signet_hash = bitcoin_cli(['getblockhash', '1'])
+    if curr_signet_hash != GLOBAL_FIRST_BLOCK_HASH:
+        print('The global faucet cannot be used with a custom Signet network. Please use the global signet or setup your custom faucet to use this functionality.\n')
+        exit(1)
+
+if args.addr == '':
+    # get address for receiving coins
+    args.addr = bitcoin_cli(['getnewaddress', 'faucet', 'bech32'])
 
 data = {'address': args.addr, 'password': args.password}
 try:
     res = requests.post(args.faucet, data=data)
 except:
     print('Unexpected error when contacting faucet:', sys.exc_info()[0])
```

Comment:
Currently, the message printed here be cryptic to a non-Python savvy user. For example, when attempting to connect to a URL with no schema (e.g. `test.com`) you get the following:

```sh
$ ./contrib/signet/getcoins.py --faucet="test.com"
Unexpected error when contacting faucet: <class 'requests.exceptions.MissingSchema'>
```

While we're at it, I'd suggest the following for a more digestible error message:

```suggestion
    print('Unexpected error when contacting faucet:', sys.exc_info()[1])
```

which results in the following message:

```sh
$ ./contrib/signet/getcoins.py --faucet="test.com"
Unexpected error when contacting faucet: Invalid URL 'test.com': No schema supplied. Perhaps you meant http://test.com?
```

**Note**: this would also affects other similar error messages, such as a `connectionError` (tested with `--faucet="http://fhjvjhj.com/"`).